tool/teleport/common/teleport.go

```diff
@@ -96,6 +96,8 @@ func Run(options Options) (app *kingpin.Application, executedCommand string, con
 		fmt.Sprintf("Comma-separated list of roles to start with [%s]", strings.Join(defaults.StartRoles, ","))).
 		Short('r').
 		StringVar(&ccf.Roles)
+	start.Flag("join-method", "Method to use to join the cluster, can be \"token\", \"ec2\", or \"iam\"").
+		EnumVar(&ccf.JoinMethod, "token", "ec2", "iam")
```

Comment:
nit: Can we extract those as constants or possibly reuse existing one? Quite often I see a new option being added without updating application help.
I think something like that is a good idea: https://github.com/gravitational/teleport/blob/55cbd0ac9747a2012ee0373c0e69c22495dce158/lib/client/identityfile/identity.go#L80-L81
As help can be automatically generated: https://github.com/gravitational/teleport/blob/55cbd0ac9747a2012ee0373c0e69c22495dce158/tool/tctl/common/auth_command.go#L103-L104